src/reflect/scala/reflect/internal/Positions.scala

```diff
@@ -345,7 +345,8 @@ trait Positions extends api.Positions { self: SymbolTable =>
           if (t.pos includes pos) {
             if (isEligible(t)) last = t
             super.traverse(t)
-          } else t match {
+          }
+          t match {
```

Comment:
I believe this was a latent bug in the presentation compiler. It became apparent in the REPL completion tests after I (mostly) removed the hack of inserting `__DUMMY__` at the cursor in this PR.

In erroneous code, the range position of the `TypeDef` of an annotated parameter can contain the annotation itself. So we should always try traversing into `.symbol.annotations.map(_.original)`.